test/res_man_test.rb

```diff
@@ -0,0 +1,223 @@
+      end
+
+      it "raises an exception if there are no available rooms for the date" do
+         20.times do
+            @phillip.make_reservation(@check_in, @check_out)
+         end
+
+         expect{ @phillip.make_reservation(@check_in, @check_out) }.must_raise ArgumentError
+      end
+
+      it "can book a room when someone else is checking out that day" do
+         reservations = []
+         reservations << @phillip.make_reservation(@check_in, @check_out)
+         reservations << @phillip.make_reservation(@check_out, @check_out + 1)
+         
+         reservations.each do |reservation|
+            expect(reservation.room.room_num).must_equal 1
+         end
+      end
+   end
+
+   describe "reservations_by_room method" do
+      before do
+         @room_num = 1
+         @start_date = Date.new(2020, 5, 30)
+         @end_date = Date.new(2020, 5, 31)
+         @reservation = @phillip.make_reservation(
+            @start_date, 
+            @end_date
+         )
+      end
+
+      it "finds reservation with room number match" do
+         matching_reservations = @phillip.reservations_by_room(
+            @room_num,
+            @start_date,
+            @end_date
+         )
+         expect(matching_reservations.length).must_equal 1
+         expect(matching_reservations[0]).must_equal @reservation
+         expect(matching_reservations[0].room.room_num).must_equal @room_num
+      end
+
+      it "finds no reservation if no room number match" do
+         matching_reservations = @phillip.reservations_by_room(
+            999,
+            @start_date,
+            @end_date
+         )
+         expect(matching_reservations.length).must_equal 0
+      end
+
+      it "finds no reservation if no date match" do
+         matching_reservations = @phillip.reservations_by_room(
+            @room_num,
+            Date.new(2019, 03, 10),
+            Date.new(2019, 03, 20),
+         )
+         expect(matching_reservations.length).must_equal 0
+      end
+   end
+
+   describe "reservations_by_date method" do
+      before do
+         @check_in = Date.new(2020, 3, 10)
+         @check_out = Date.new(2020, 3, 11)
+         @reservation = @phillip.make_reservation(
+            @check_in, 
+            @check_out
+         )
+      end
+
+      it "can retrieve a reservation given a valid date" do
+         matching_reservations = @phillip.reservations_by_date(@check_in)
+         
+         expect(matching_reservations.length).must_equal 1
+         expect(matching_reservations[0]).must_equal @reservation
+      end
+
+      it "will not retrieve any reservations for a non-matching date" do
+         matching_reservations = @phillip.reservations_by_date(
+            Date.new(2019, 03, 20)
+         )
+         
+         expect(matching_reservations.length).must_equal 0
+      end
+   end 
+
+   describe "view_by_vacancy method" do
+      it "retrieves all rooms for a range with no bookings" do
```

Comment:
I am a bit confused why the `check_in` and `check_out` date are the same in this test. To ensure that `it retrieves all rooms for a range with no bookings` you should make sure to test the method with a valid date range.